plasmapy/plasma/fields.py

```diff
@@ -0,0 +1,700 @@
+            return np.mean(np.gradient(self.xaxis))
+        else:
+            raise ValueError(
+                "dx property is only well-defined for " "uniformly-spaced grids."
+            )
+
+    @property
+    def dy(self):
+        """
+        Calculate dy (only valid for a uniform grid)
+        """
+        if self._regular_grid is True:
+            return np.mean(np.gradient(self.yaxis))
+        else:
+            raise ValueError(
+                "dy property is only well-defined for " "uniformly-spaced grids."
+            )
+
+    @property
+    def dz(self):
+        """
+        Calculate dz (only valid for a uniform grid)
+        """
+        if self._regular_grid is True:
+            return np.mean(np.gradient(self.zaxis))
+        else:
+            raise ValueError(
+                "dz property is only well-defined for " "uniformly-spaced grids."
+            )
+
+    @property
+    def shape(self):
+        """
+        Shortcut method for retriving the shape of the grid
+        """
+        return self.grid.shape
+
+    @property
+    def unit(self):
+        """
+        Shortcut method for retriving the unit of the grid
+        """
+        return self.grid.unit
+
+    def _make_grid(self, **kwargs):
+        """
+        Create a grid based on keywords set
+        """
+
+        if "num" in kwargs.keys():
+            num = kwargs["num"]
+        else:
+            num = 100
+
+        if "length" in kwargs.keys():
+            length = kwargs["length"]
+        else:
+            length = 1 * u.cm
+
+        if "regular_grid" in kwargs.keys():
+            regular_grid = kwargs["regular_grid"]
+        else:
+            regular_grid = True
+
+        # Process different options for inputs
+        if isinstance(num, int):
+            num = (num, num, num)
+
+        L = np.zeros([2, 3]) * length.unit
+        if length.size == 1:
+            L[:, 0] = np.array([-length.value, length.value]) * length.unit
+            L[:, 1] = np.array([-length.value, length.value]) * length.unit
+            L[:, 2] = np.array([-length.value, length.value]) * length.unit
+        elif length.size == 3:
+            L[:, 0] = np.array([-length[0].value, length[0].value]) * length.unit
+            L[:, 1] = np.array([-length[1].value, length[1].value]) * length.unit
+            L[:, 2] = np.array([-length[2].value, length[2].value]) * length.unit
+        else:
+            raise ValueError("Dimensions of length in _create_grid are not valid.")
+
+        # Create a blank grid
+        grid = np.zeros([num[0], num[1], num[2], 3]) * u.cm
+
+        # If regulr_grid keyword is False, create a non-uniform grid with
+        # random point spacing in all directions
+        if not regular_grid:
+            for d in [0, 1, 2]:
+                ax = np.random.uniform(low=L[0, d].value, high=L[1, d].value, size=num)
+                ax = np.sort(ax, axis=d)
+                grid[..., d] = ax * L.unit
```

Comment:
This is a pretty important option for testing the proton radiography module: irregularly gridded fields are a commonly produced by Lagrangian fluid simulations, (or Eulerian codes with adaptive mesh refinement), so the ability to use that output without computationally expensive re-gridding is very helpful. If you have a better way of generating such a mesh I'm all ears though: this was a quick way of doing it. 

Implementing dimensions is a good idea. I'm not entirely clear what implementing non-Cartesian coordinate systems would entail: can you elaborate on that?

Reply:
So, implementing non-Cartesian coordinate systems would involve a substantial change: self._grid could not be a u.Quantity because the different axes would have different units.

Reply:
> So, implementing non-Cartesian coordinate systems would involve a substantial change: self._grid could not be a u.Quantity because the different axes would have different units.

Fair.  However I feel the discuss is till warranted. But the will be in the future for another PR.

Reply:
> This is a pretty important option for testing the proton radiography module: irregularly gridded fields are a commonly produced by Lagrangian fluid simulations, ...

I see.  I'll need to think about this more then.  My first concern is with using `np.sort` because you risk shrinking `ax`.  `np.sort` performs a sort and a unique operation.  This is not my forte, but @namurphy might have some good suggestions.

Reply:
I tested this and it seems to be that `np.sort` does not return a unique list? I'm currently still using that over in #909 but open to other suggestions.

Reply:
@pheuer You're right. I was mistaken.